include/render/gles2.h

```diff
@@ -32,11 +32,37 @@ struct wlr_gles2_renderer {
 	const char *exts_str;
 
 	struct {
-		GLuint quad;
-		GLuint ellipse;
-		GLuint tex_rgba;
-		GLuint tex_rgbx;
-		GLuint tex_ext;
+		struct {
+			GLuint program;
+			GLint proj;
+			GLint color;
+		} quad;
+		struct {
+			GLuint program;
+			GLint proj;
+			GLint color;
+		} ellipse;
+		struct {
```

Comment:
Since this struct is shared between `tex_rgba`, `tex_rgbx` and `tex_ext`, we can probably use the same `struct tex_shader` for these three shaders?